src/commands/dev.ts

```diff
@@ -170,12 +171,12 @@ const sendError = (res, status) => {
 function getUrlFromFile(mountedDirectories: [string, string][], fileLoc: string): string | null {
   for (const [dirDisk, dirUrl] of mountedDirectories) {
     if (fileLoc.startsWith(dirDisk + path.sep)) {
-      const fileExt = path.extname(fileLoc).substr(1);
+      const {baseExt} = getExt(fileLoc);
       const resolvedDirUrl = dirUrl === '/' ? '' : dirUrl;
       return fileLoc
         .replace(dirDisk, resolvedDirUrl)
         .replace(/[/\\]+/g, '/')
-        .replace(new RegExp(`${fileExt}$`), srcFileExtensionMapping[fileExt] || fileExt);
+        .replace(new RegExp(`\\${baseExt}$`), srcFileExtensionMapping[baseExt] || baseExt);
```

Comment:
why the `\\`?

Reply:
A single `\` escapes the `$` next to it so it’s no longer escaping that variable.

Also we this to become `/\.js$`, not `.js$`, so we match `.js` but not `.mjs`.

Reply:
Would it make sense to move the regex creation into the `getExt` function so we can document it and not repeat it?

Reply:
@stramel I like that idea!

There were actually so many other places where we could reduce duplication between dev, build, and install, but I just capped it at a reviewable PR length for now. I’ll definitely keep that idea in mind as I make more cleanup PRs that just (hopefully) reduce complexity without affecting functionality, but first I want to see how small we can get some of these checks.

Agree, though—we should be doing a lot less on-the-fly RegEx matching if we can help it. But I want to get more areas cleaned up first.